include/chemfiles/File.hpp

```diff
@@ -53,8 +69,9 @@ class CHFL_EXPORT File {
 /// the user of the class to the current state.
 class CHFL_EXPORT TextFile: public File, public std::iostream {
 public:
-    /// Open the most adaptated text file class for the given `path` and `mode`
-    static std::unique_ptr<TextFile> create(const std::string& path, File::Mode mode);
+    /// Open the file at the given `path` with the requested `mode` and
+    /// `compression` method using the most adapted child class of `TextFile`.
+    static std::unique_ptr<TextFile> open(std::string path, File::Mode mode, File::Compression compression);
```

Comment:
Would a default argument for compression (`DEFAULT`) be useful here?

Reply:
I don't think this is needed, as only Trajectory will call this function (through format_creator_t lambda, through FormatFactory), and it will always do so with all parameters.

Reply:
I see, it was the first thing I saw without reading the rest of the PR.